containers/blockchain/blockchainNetwork/chaincode/src/bcfit/contract.go

```diff
@@ -53,6 +52,32 @@ func (t *SimpleChaincode) makePurchase(stub shim.ChaincodeStubInterface, args []
 	}
 	contract.Quantity = quantity
 
+	contract.Id = strings.ToLower(args[4])
+
+	//contract id must be in format c and then six digits i.e c123456
+	contractIDarray := strings.SplitAfter(contract.Id, "c")
+	if len(contractIDarray) != 2 {
+		return shim.Error("Incorrect contract ID format")
+	} else if contractIDarray[0] != "c" {
+			return shim.Error("Incorrect contract ID format")
+	} else if len(contractIDarray[1]) != 6 {
+			return shim.Error("Incorrect contract ID format")
+	} else if _, err := strconv.Atoi(contractIDarray[1]); err != nil {
+			return shim.Error("Incorrect contract ID format")
+	}
+
+	//check if contract id already exists
+	var existingContract Contract
+	existingContractAsBytes, err := stub.GetState(contract.Id)
+	if err != nil {
+		return shim.Error("Failed getting constract state")
```

Comment:
@raheelzubairy Just need some clarity on this one:
What happens to `stub.GetState(contract.Id)` if the contract ID isn't existing (which it shouldn't since we're making a new one).
Will `err != nil` be true? That would mean the function will always throw this error, wouldn't it?